modules/_node-scripts/format/index.mjs

```diff
@@ -0,0 +1,309 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {ESLint} from 'eslint';
+import fg from 'fast-glob';
+import fs from 'fs';
+import path from 'path';
+import prettier from 'prettier';
+import stylelint from 'stylelint';
+
+import {getRootDir} from '../util/constants.mjs';
+import filterChangedFiles from '../util/filterChangedFiles.mjs';
+import {readIgnoreFile} from '../util/readIgnoreFile.mjs';
+import {ID_END, ID_START} from './jsp/getPaddedReplacement.mjs';
+import processJSP from './jsp/processJSP.mjs';
+import {SCRIPTLET_CONTENT} from './jsp/substituteTags.mjs';
+import {BLOCK_CLOSE, BLOCK_OPEN} from './jsp/tagReplacements.mjs';
+import {FILLER_CHAR, SPACE_CHAR, TAB_CHAR} from './jsp/toFiller.mjs';
+import logError from './logError.mjs';
+
+const PRETTIER_IGNORE_FILE = '.prettierignore';
+const ESLINT_IGNORE_FILE = '.eslintignore';
+const GIT_IGNORE_FILE = '.gitignore';
+
+async function getFilesToCheck() {
+	const rootDir = await getRootDir();
+
+	const eslintIgnoreFilePath = path.join(rootDir, ESLINT_IGNORE_FILE);
+	const prettierIgnoreFilePath = path.join(rootDir, PRETTIER_IGNORE_FILE);
+	const gitIgnoreFilePath = path.join(rootDir, GIT_IGNORE_FILE);
+
+	const eslintIgnores = readIgnoreFile(eslintIgnoreFilePath);
+	const prettierIgnores = readIgnoreFile(prettierIgnoreFilePath);
+	const gitIgnores = readIgnoreFile(gitIgnoreFilePath);
+
+	return await fg(
+		[
+			'**/types/**/*.d.ts',
+			'*.{js,mjs,ts}',
+			'**/*.{css,js,mjs,scss,ts,tsx}',
+			'**/src/**/*.{jsp,jspf}',
+		],
+		{
+			dot: true,
+			ignore: [
+				'**/src/test/**',
+				'**/build_gradle/**',
+				...gitIgnores,
+				...eslintIgnores,
+				...prettierIgnores,
+			].map((ignore) => {
+				if (ignore.startsWith('*') && !ignore.startsWith('**')) {
+					ignore = `**/${ignore}`;
+				}
+
+				if (!ignore.startsWith('*')) {
+					ignore = `**${ignore.startsWith('/') ? '' : '/'}${ignore}`;
+				}
+
+				if (!ignore.endsWith('**') && !ignore.includes('.')) {
+					ignore = `${ignore}${ignore.endsWith('/') ? '' : '/'}**`;
+				}
+
+				return ignore;
+			}),
+		}
+	);
+}
+
+const FALLBACK_FILE_PATH = '__fallback__.js';
+
+export default async function format() {
+	const fix = process.argv[3] !== '--check';
+
+	let filepaths = await getFilesToCheck();
+
+	filepaths = await filterChangedFiles(filepaths);
+
+	const eslintConfig = await getEslintConfig();
```

Comment:
You can read these 3 in parallel by using `await Promise.all(...` 
I do it in several places and once you get used to the pattern it's very intuitive. And of course it makes the build much faster.